tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -91,62 +96,135 @@ var _ = Describe("nftables", Ordered, Label(tsparams.LabelNftablesTestCases), Co
 		Expect(err).ToNot(HaveOccurred(), "Failed to create route map with specific routes")
 
 		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "add", hubIPv4Network)
+
+		By("Setup test environment")
+		masterPod = setupRemoteMultiHopTest(ipv4SecurityIPList, hubIPv4ExternalAddresses,
+			ipv4NodeAddrList, cnfWorkerNodeList, masterNodeList)
 	})
 
 	AfterAll(func() {
 		By("Remove the static route to the external Pod network on each worker node")
 		addDeleteStaticRouteOnWorkerNodes(testPodList, routeMap, "del", hubIPv4Network)
 
-		By("Disables nftables in cluster")
-		disableNftablesIfActive()
+		By("Disables nftables on worker-0 if active")
+		disableNftablesIfActive(cnfWorkerNodeList[0].Definition.Name)
+
+		By("Remove machine-configuration")
+		err := mco.NewMCBuilder(APIClient, mcNftablesName).Delete()
+		Expect(err).ToNot(HaveOccurred(), "Failed to get the machineConfig")
+
+		err = netenv.WaitForMcpStable(APIClient, 35*time.Minute, 1*time.Minute, NetConfig.CnfMcpLabel)
+		Expect(err).ToNot(HaveOccurred(), "Failed to wait for MCP to be stable")
 	})
 
 	Context("custom firewall", func() {
 		AfterEach(func() {
 			By("Define and delete a NFTables custom rule")
 			createMCAndWaitforMCPStable(tsparams.CustomFirewallDelete, mcNftablesName)
+
+			By("Remove machine-configuration")
 			err := mco.NewMCBuilder(APIClient, mcNftablesName).Delete()
 			Expect(err).ToNot(HaveOccurred(), "Failed to get the machineConfig")
+
+			err = netenv.WaitForMcpStable(APIClient, 35*time.Minute, 80*time.Second, NetConfig.CnfMcpLabel)
```

Comment:
Just curious, why 80 time*second?

Reply:
when the machineConfig is deleted it can take longer then a min for the mcp to be stable and the nftables custom firewall to be removed from the node.